Refactor/confirmation modal

printgourav opened the pull request:
Fixes Issue: #714 

## Description

This PR moves/refactors common modal logic from `DeleteAccountConfirmationModal, DeleteKeyModal` components to new component named `ConfirmationModal.jsx`. Thus, removing duplicate logic

- added new component `ConfirmationModal.jsx`
- highlighted api-key-name (primary-color) in `Dashboard.jsx` api-key modal
- removed unnecessary/repeated css from `SettingCard.jsx` & `Dashboard.jsx`
- updated modal test-id in `SettingCard.jsx` & `Dashboard.jsx`
- added `type` attribute in `Button` component
- added missing dev-dependency `@testing-library/user-event`
- added test-case for `ConfirmationModal.jsx`

## What type of PR is this? (Check all applicable)

- [x] 🍕 Feature
- [x] 👨‍💻 Code Refactor
- [x] ✅ Test

## Screenshots (if applicable)

### Test case

![image](https://github.com/user-attachments/assets/4fbe0cee-1fbe-4d8f-9710-4517ea0c138e)

### Test Coverage

![image](https://github.com/user-attachments/assets/02e7d8be-6c7a-4389-8d56-f0bc7c4e50d7)

![image](https://github.com/user-attachments/assets/41e6f18b-c57f-469e-af3f-e6e17e5c204f)

![image](https://github.com/user-attachments/assets/379364bb-47ec-4a1b-a75b-a88ec6aaaa6b)

### Demo

https://github.com/user-attachments/assets/759c2ded-60e7-4163-a4da-91f0f9a914a1

## Checklist
- [x] I have performed a self-review of my code 
- [x] I have added tests that prove my fix is effective or that my feature works  
- [x] New and existing unit tests pass locally with my changes

printgourav committed:
refactor: delete account flow with new confirm modal

printgourav committed:
refactor: delete account flow with new confirm modal

printgourav committed:
refactor: delete account flow with new confirm modal

Allan2000-Git commented:
@printgourav Resolve the issue pointed by Copilot

TanishqSingla commented on packages/ui/src/utils/Constants.js:
```diff
@@ -648,14 +652,21 @@ export const VERIFICATION = {
 };
 
-export const DELETE_ACCOUNT_CONFIRMATION_MODAL = {
+export const DELETE_ACCOUNT = {
```
```suggestion
export const DELETE_ACCOUNT_MODAL = {
```

TanishqSingla commented on packages/ui/src/components/settings/SettingCard.module.css:
```diff
@@ -19,3 +19,16 @@
   width: 100%;
 }
+.delete-account-modal-title {
+  margin-bottom: 15px;
```
Can you confirm if you're using `15px` value in other modals?

TanishqSingla commented on packages/ui/src/components/confirm/ConfirmationModal.jsx:
```diff
@@ -0,0 +1,66 @@
+    <Modal data-testid="confirmation-modal" onClose={onClose} isOpen={isOpen}>
+      <div>
+        <h2
+          className={`${styles["confirm-modal-heading"]}, ${headingClassName}`}
```
please fix this

TanishqSingla commented on packages/ui/__tests__/components/SettingCard.test.jsx:
```diff
@@ -40,4 +95,85 @@ describe("SettingCard Component", () => {
+    renderWithProviders();
+    fireEvent.click(screen.getByText(/delete account/i));
+    expect(
+      screen.getByText(/are you sure you want to delete/i)
```
`getByText` is itself an assertion i.e it will fail if the component is not found. This makes `expect` redundant

theboyofdream committed:
Merge branch 'stage' of https://github.com/TeamShiksha/openlogo into refactor/confirmation-modal

theboyofdream committed:
feat(ui): confirmation popup modal
- added `type` attribute in `Button.jsx`
- added test case for `ConfirmationModal.jsx`
- installed missing dev-dependency `@testing-library/user-event`
- removed unnecessary css for confirmation modal from `Dashboard.module.css` & `SettingCard.module.css`
- update modal test-id in `Dashboard.jsx` & `SettingCard.jsx`

TanishqSingla commented on packages/ui/src/components/confirm/ConfirmationModal.jsx:
```diff
@@ -0,0 +1,63 @@
+
+  return (
+    <Modal onClose={onClose} isOpen={isOpen}>
+      <form noValidate onSubmit={handleSubmit}>
```
There are no input tags to validate so why a div, and the buttons just use a handler with `preventDefault` just use a div

TanishqSingla commented on packages/ui/src/components/common/button/Button.jsx:
```diff
@@ -6,6 +6,7 @@ import LoadingSpinner from "../loadingspinner/LoadingSpinner";
+  type = "button",
   onClick = () => {},
   disabled = false,
   className = "",
```
instead of adding native html props manually with their type, use `...props` technique to add default props and add a jsdoc for autocomplete

TanishqSingla commented on packages/ui/__tests__/page/dashboard/Dashboard.test.jsx:
```diff
@@ -28,11 +36,12 @@ const mockToastContext = {
 const mockUserContext = (userData, loading) => ({
   userData,
-  loading,
+  loading: loading,
```
why?

TanishqSingla commented on packages/ui/__tests__/page/dashboard/Dashboard.test.jsx:
```diff
@@ -20,6 +21,13 @@ import {
 
+vi.mock("../../src/hooks/useApi.js", () => ({
+  useApi: vi.fn(() => ({
+    makeRequest: vi.fn(() => Promise.resolve(true)),
```
nit: use `vi.fn()` so later it's implementation can be modified in the test

TanishqSingla commented on packages/ui/src/page/dashboard/Dashboard.jsx:
```diff
@@ -194,16 +223,38 @@ function Dashboard() {
+          <ConfirmationModal
+            isOpen={showModal}
+            onClose={handleModalClose}
+            onConfirm={handleDeleteKey}
```
```suggestion
            onConfirm={handleDelete}
```

TanishqSingla commented on packages/ui/src/page/dashboard/Dashboard.jsx:
```diff
@@ -58,9 +65,40 @@ function Dashboard() {
+    setConfirmKeyName(e.target.value);
+  };
+
+  const handleDeleteKey = async () => {
```
This function implies it handles action on pressing delete key on keyboard.
```suggestion
  const handleDelete = async () => {
```

TanishqSingla commented on packages/ui/package.json:
```diff
@@ -27,6 +27,7 @@
   },
   "devDependencies": {
     "@eslint/js": "^9.11.1",
+    "@testing-library/user-event": "^14.6.1",
```
is this really needed?

theboyofdream committed:
fix: address PR review comments
- removed "@testing-library/user-event" package because we can use `fireEvent` from `@testing-library/react`
- updated `SettingCard` test-case
    - replaced static text with constants (imported from `Constants.js`)
- `Button.jsx` removed type prop & using `...rest` to accept props of button
- removed extra `div` wrapped around children & novalidate attribute of form from `ConfirmationModal`

theboyofdream committed:
fix: include missed changes from last commit

TanishqSingla commented on package.json:
```diff
@@ -22,6 +22,7 @@
   },
   "devDependencies": {
     "@eslint/js": "^9.11.1",
+    "@testing-library/user-event": "^14.6.1",
```
this is already present in the UI package, why add in root?
